File: core-java-practice/leet-code-codebase/PalindromeNumber.java
import java.util.*;
public class palindrome_no {
  public static void main(String[] args) {
    Scanner sc=new Scanner(System.in);
    int n=sc.nextInt();
    int rv=0;
    int duplicate=n;
    while (n>0) {
      int ld=n%10;
      rv=rv*10+ld;
      n=n/10;
    }
    //matching if the original and the reverse ones are same or not 
    if(rv==duplicate){
      System.out.println("YES");
    }else{
      System.out.println("NO");
    }
  }
}

File: core-csharp-practice/leet-code-codebase/PalindromeNumber.cs
using System;
namespace ProgrammingElements{
    class PalindromeNo{
        static void Main(string[] args){
            int n = int.Parse(Console.ReadLine());
            int rv = 0;
            int duplicate = n;
            while (n > 0){
                int ld = n % 10;
                rv = rv * 10 + ld;
                n = n / 10;
            }
            // matching if the original and the reverse numbers are same
            if (rv == duplicate){
                Console.WriteLine("YES");
            }
            else{
                Console.WriteLine("NO");
            }
        }
    }
}
